### PR TITLE
TreeNode added, Tree interface added, Tree insert and traversal opera…

### PR DESCRIPTION
**Pull request by sougatabhattacharjee**
- TreeNode extended, 
- Tree interface added
- Insert operation extended 
- Tree traversal methods added
- Basic test case for insert added

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -11,5 +11,7 @@ repositories {
 
 dependencies {
     compile 'org.apache.commons:commons-lang3:3.4'
+    compile group: 'com.google.guava', name: 'guava', version: '11.0.2'
     testCompile group: 'junit', name: 'junit', version: '4.11'
+
```
Comment (sauravbiswasiupr) [inline build.gradle]: remove extra line

File: src/main/java/BST/Tree.java [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
-public class Tree {
-    public TreeNode root;
-    public Tree() {}
+public interface Tree<Data extends Comparable<?>> {
 
-    public Tree(TreeNode node) {
-        root = node;
-    }
+    /**
+     * Add a node the tree with value n. Its O(lgn)
```
Comment (sauravbiswasiupr) [inline src/main/java/BST/Tree.java]: node in the tree. Please change wherever applicable.

File: src/main/java/BST/TreeImpl.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    public enum Traverse_Order {
+        PREORDER,
+        INORDER,
+        POSTORDER
+    }
+
+    public void insert(final Comparable data) {
+        Preconditions.checkNotNull(data, "Null or Empty node cannot be inserted");
+
+        this.root = insert((Data) data, this.root);
```
Comment (sauravbiswasiupr) [inline src/main/java/BST/TreeImpl.java:28]: 👍

File: build.gradle (hunk 1 above)
Comment (sougatabhattacharjee) [inline build.gradle]: done, but forgot to merge the previous changes..the junit version should be 5.x.x

File: src/main/java/BST/Tree.java (hunk 2 above)
Comment (sougatabhattacharjee) [inline src/main/java/BST/Tree.java]: done

**Comment by sauravbiswasiupr** [comment]
🚀 🎸

**Comment by sougatabhattacharjee** [comment]
merged feature/treeNode branch

**Pull request merged**